packages/transaction-controller/src/TransactionController.ts

```diff
@@ -3072,10 +3081,41 @@ export class TransactionController extends BaseControllerV1<
     );
   }
 
+  async #updateTransactionLayer1GasFee(
```

Comment:
To avoid adding more logic to the main controller file, would this be a great candidate for a public method in `utils/layer-1-gas-fee-flows.ts`?

It also overlaps with the new logic in `GasFeePoller` so we could call it from there too to remove the duplication. We could return the value rather than assigning to the meta directly.

Plus it would help isolate any unit tests it needs.